go/vt/vtgate/engine/vstream.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2021 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package engine
+
+import (
+	"fmt"
+	"io"
+
+	vtrpcpb "vitess.io/vitess/go/vt/proto/vtrpc"
+	"vitess.io/vitess/go/vt/vterrors"
+
+	"vitess.io/vitess/go/sqltypes"
+	"vitess.io/vitess/go/vt/key"
+	"vitess.io/vitess/go/vt/vtgate/vindexes"
+
+	binlogdatapb "vitess.io/vitess/go/vt/proto/binlogdata"
+	querypb "vitess.io/vitess/go/vt/proto/query"
+)
+
+var _ Primitive = (*VStream)(nil)
+
+// VStream is an operator for streaming specific keyspace, destination
+type VStream struct {
+	Keyspace          *vindexes.Keyspace
+	TargetDestination key.Destination
+	TableName         string
+	Position          string
+	Limit             int
+
+	noTxNeeded
+	noInputs
+}
+
+// RouteType implements the Primitive interface
+func (v *VStream) RouteType() string {
+	return "VStream"
+}
+
+// GetKeyspaceName implements the Primitive interface
+func (v *VStream) GetKeyspaceName() string {
+	return v.Keyspace.Name
+}
+
+// GetTableName implements the Primitive interface
+func (v *VStream) GetTableName() string {
+	return v.TableName
+}
+
+// Execute implements the Primitive interface
+func (v *VStream) Execute(_ VCursor, _ map[string]*querypb.BindVariable, wantfields bool) (*sqltypes.Result, error) {
+	return nil, vterrors.New(vtrpcpb.Code_UNIMPLEMENTED, "'Execute' in VStream is not supported")
```

Comment:
This should be `vtrpcpb.Code_INTERNAL`, since we'll never implement Execute on vstream, right?